core-api/src/main/java/com/optimizely/ab/config/audience/UserAttribute.java

```diff
@@ -45,25 +49,31 @@ public String getType() {
         return type;
     }
 
+    public String getMatch() {
+        return match;
+    }
+
     public Object getValue() {
         return value;
     }
 
-    public boolean evaluate(Map<String, ?> attributes) {
+    public @Nullable Boolean evaluate(Map<String, ?> attributes) {
         // Valid for primative types, but needs to change when a value is an object or an array
         Object userAttributeValue = attributes.get(name);
 
-        if (value != null) { // if there is a value in the condition
-            // check user attribute value is equal
-            return value.equals(userAttributeValue);
+        if (!"custom_attribute".equals(type)) {
+            MatchType.logger.error(String.format("condition type not equal to `custom_attribute` %s", type != null ? type : ""));
+            return null; // unknown type
         }
-        else if (userAttributeValue != null) { // if the datafile value is null but user has a value for this attribute
-            // return false since null != nonnull
-            return false;
+        // check user attribute value is equal
+        try {
+            return MatchType.getMatchType(match, value).getMatcher().eval(userAttributeValue);
```

Comment:
Based on the name alone, I find it surprising that we're passing `value` to `.getMatchType(...)` and not to `.getMatcher(...)` or `.eval(...)`. That said, this is minor and I haven't looked at those functions yet.